### PR TITLE
System tests for assignments

### PR DESCRIPTION
**Pull request by MrBartusek**
~~**NOTE:** This PR is not completed yet cause #778 is not merged. When new assignments page will be done I will add testing for _element restriction_ and _datetimepicker_ too!~~

As I promised these tests are finally done!  New tests in this PR:
- `when user is mentor should create assignment`
- `when user is mentors should be able to edit assignment`
- `when user is a member should be able to make assignment project`

Additional changes:
-  Fixed some problems with create assignment page
- Removed `popper.js` cause it was breaking `application.js` and makes these tests to fail https://github.com/CircuitVerse/CircuitVerse/pull/811/commits/65919110375cea38ef3d12a613e8564d62f9a2aa

This is a [GCI Task](https://codein.withgoogle.com/dashboard/task-instances/4958246641598464/)

### CODE REVIEW

File: spec/rails_helper.rb [hunk 1]
```diff
@@ -40,7 +40,7 @@
 RSpec.configure do |config|
   # Remove this line if you're not using ActiveRecord or ActiveRecord fixtures
   config.fixture_path = "#{::Rails.root}/spec/fixtures"
-  config.include Devise::Test::IntegrationHelpers, type: :request
+  config.include Devise::Test::IntegrationHelpers
```
Comment (tachyons) [inline spec/rails_helper.rb]: May be you can add the same module for both request and system, not for all tests

File: spec/system/assignment_spec.rb [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+describe "Assignments", type: :system do
+  before do
+    @mentor = FactoryBot.create(:user)
+    @group = FactoryBot.create(:group, mentor: @mentor)
+    @member = FactoryBot.create(:user)
+    FactoryBot.create(:group_member, group: @group, user: @member)
+  end
+
+  describe "user is mentor" do
```
Comment (tachyons) [inline spec/system/assignment_spec.rb]:
```suggestion
  context "when user is mentor" do
```

File: spec/system/assignment_spec.rb [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      fill_assignments(name, deadline, description, grading: true)
+
+      click_button "Create Assignment"
+
+      expect(page).to have_text("Assignment was successfully created.")
+      expect(page).to have_text(name)
+      # expect(page).to have_text(deadline.strftime("%d %b,%Y"))
+    end
+
+    it "should edit assignment" do
```
Comment (tachyons) [inline spec/system/assignment_spec.rb]:
```suggestion
    it "should be able to edit assignment" do
```

File: spec/system/assignment_spec.rb [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+      click_button "Update Assignment"
+
+      expect(page).to have_text("Assignment was successfully updated.")
+      expect(page).to have_text(name)
+      # expect(page).to have_text(deadline.strftime("%d %b,%Y"))
+    end
+  end
+
+  describe "user is member" do
```
Comment (tachyons) [inline spec/system/assignment_spec.rb]:
```suggestion
  context "when user is a member" do
```

**Comment by MrBartusek** [comment]
Edited description  🚀

**Comment by tachyons** [comment]
Only group creation is included in this PR so far, Need group managing group feature as well, like editing, deleting, adding members etc

**Comment by MrBartusek** [comment]
@tachyons this pr is only fo assignments, someone is working on group tests on #814

**Comment by tachyons** [comment]
Sorry, I commented on wrong PR

**Comment by tachyons** [comment]
Can you rebase and make tests pass for this PR ?

**Comment by MrBartusek** [comment]
Done 🚀 @tachyons

**Comment by tachyons** [comment]
Awesome

**Pull request merged**